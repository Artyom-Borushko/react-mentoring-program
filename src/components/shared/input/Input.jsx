import './input.css';
import React, {useState} from "react";
import Button from "../button/Button";

export default function Input({labelText, id, type, initialValue = '', dropdownOptions = [], ref, dropdownOpenerText = ''}) {

    const [showDropdown, setShowDropdown] = useState(false);

    const toggleDropdown = (event) => {
        event.preventDefault();
        setShowDropdown(!showDropdown);
    };

    switch (type) {
        case 'checkbox':
            return (
                <div className={'input-field'}>
                    {/*TODO - probably can modify and reuse already existing OptionsSelector to work with checkboxes*/}
                    <Button
                        type={'dropdown-opener'}
                        onClick={toggleDropdown}
                    >
                        {dropdownOpenerText}
                    </Button>
                    {showDropdown && dropdownOptions.map((option) => (
                        <label key={option}>
                            <input
                                id={id}
                                type={type}
                                name={id}
                                defaultValue={option}
                                ref={ref}
                            />
                            {option}
                        </label>
                    ))}
                </div>
            )
        case 'textarea':
            return (
                <div className={'input-field textarea'}>
                    <label
                        htmlFor={id}
                    >
                        {labelText}
                    </label>
                    <textarea
                        className={type}
                        id={id}
                        name={id}
                        defaultValue={initialValue}
                        ref={ref}
                    >
                    </textarea>
                </div>
            )
        default:
            return (
                <div className={'input-field'}>
                    <label
                        htmlFor={id}
                    >
                        {labelText}
                    </label>
                    <input
                        className={labelText.toLowerCase()}
                        type={type}
                        id={id}
                        name={id}
                        defaultValue={initialValue}
                        ref={ref}
                    />
                </div>
            )
    }
}
