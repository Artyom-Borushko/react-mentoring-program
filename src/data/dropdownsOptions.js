export const movieDropdownOptions = ['edit', 'delete'];

export const moviesSortDropdownOptions = ['release date', 'title'];
