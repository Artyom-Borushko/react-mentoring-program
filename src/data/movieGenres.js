
export const listOfGenres = ['all', 'documentary', 'comedy', 'horror', 'crime'];
