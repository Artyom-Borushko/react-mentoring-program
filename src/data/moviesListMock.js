export const listOfMoviesMock = [
    { src: '/images/movies-list/pulpFiction.png', title: "Pulp Fiction", releaseYear: "2004", relevantGenres: ['action', 'crime'], rating: '9.2', duration: '2h 22min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId1" },
    { src: '/images/movies-list/bohemianRhapsody.png', title: "Bohemian Rhapsody", releaseYear: "2003", relevantGenres: ['documentary', 'drama'], rating: '8.4', duration: '1h 27min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId2" },
    { src: '/images/movies-list/killBillVol2.png', title: "Kill Bill: Vol 2", releaseYear: "1994", relevantGenres: ['comedy', 'crime', 'oscar'], rating: '6.2', duration: '3h 12min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId3" },
    { src: '/images/movies-list/avengersWarOfInfinity.png', title: "Avengers: War of Infinity", releaseYear: "2004", relevantGenres: ['action', 'adventure'], rating: '8.2', duration: '0h 56min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId4" },
    { src: '/images/movies-list/inception.png', title: "Inception", releaseYear: "2003", relevantGenres: ['action', 'adventure', 'horror'], rating: '10.0', duration: '1h 39min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId5" },
    { src: '/images/movies-list/reservoirDogs.png', title: "Reservoir dogs", releaseYear: "1994", relevantGenres: ['crime', 'oscar'], rating: '7.7', duration: '2h 00min', description: 'Lorem ipsum dolor sit amet. Eum autem quod aut dolorem voluptatum nam dolores sunt. Ea velit placeat ea corporis illo rem quod nulla. Et maxime placeat ut vero placeat et harum sequi. A Quis expedita id dolores sint id repellendus atque sit suscipit obcaecati vel quia galisum quo cumque internos!', id: "uniqueMovieId6" },
];
