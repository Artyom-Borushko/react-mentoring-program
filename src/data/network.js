export default 'http://localhost:4000/movies';
