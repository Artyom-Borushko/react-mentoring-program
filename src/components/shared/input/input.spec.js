import {render} from "@testing-library/react";
import React from "react";
import '@testing-library/jest-dom';
import {listOfMoviesMock} from "../../../data/moviesListMock";
import Input from "./Input";

describe("Input component tests", () => {

    it("renders checkbox input", () => {
        const {container} = render(<Input
            labelText={'Genre'}
            id={'movieGenre'}
            type={'checkbox'}
            dropdownOptions={listOfMoviesMock[2].relevantGenres}
            dropdownOpenerText={'Select Genre'}
        />);

        expect(container).toMatchSnapshot();
    });

    it("renders textarea input", () => {
        const {container} = render(<Input
            labelText={'Overview'}
            id={'movieOverview'}
            type={'textarea'}
            initialValue={listOfMoviesMock[2].overview}
        />);

        expect(container).toMatchSnapshot();
    });

    it("renders text input", () => {
        const {container} = render(<Input
            labelText={'Title'}
            id={'movieTitle'}
            type={'text'}
            initialValue={listOfMoviesMock[2].title}
        />);

        expect(container).toMatchSnapshot();
    });
});
