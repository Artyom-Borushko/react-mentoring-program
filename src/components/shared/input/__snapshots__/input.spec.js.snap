// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component tests renders checkbox input 1`] = `
<div>
  <div
    class="input-field"
  >
    <button
      class="button dropdown-opener"
      type="dropdown-opener"
    >
      Select Genre
    </button>
  </div>
</div>
`;

exports[`Input component tests renders text input 1`] = `
<div>
  <div
    class="input-field"
  >
    <label
      for="movieTitle"
    >
      Title
    </label>
    <input
      class="title"
      id="movieTitle"
      name="movieTitle"
      type="text"
      value="Star Wars: The Last Jedi"
    />
  </div>
</div>
`;

exports[`Input component tests renders textarea input 1`] = `
<div>
  <div
    class="input-field textarea"
  >
    <label
      for="movieOverview"
    >
      Overview
    </label>
    <textarea
      class="textarea"
      id="movieOverview"
      name="movieOverview"
    >
      Rey develops her newly discovered abilities with the guidance of Luke Skywalker, who is unsettled by the strength of her powers. Meanwhile, the Resistance prepares to do battle with the First Order.
    </textarea>
  </div>
</div>
`;
